src/libs/ajax/DatasetBuilder.ts

```diff
@@ -62,30 +88,103 @@ export interface DatasetBuilderType {
 
 export type DatasetBuilderValue = DatasetBuilderType;
 
+export type ValueSet = {
+  domain: string;
+  values: DatasetBuilderValue[];
+};
+
+export type ValueSetApi = {
+  name: string;
+  values: string[];
+};
+
 export interface GetConceptsResponse {
   result: Concept[];
 }
 
-type DatasetRequest = {
+export type DatasetRequest = {
   cohorts: Cohort[];
   conceptSets: ConceptSet[];
-  valuesSets: { domain: string; values: DatasetBuilderValue[] }[];
+  valueSets: ValueSet[];
+};
+
+export type DatasetRequestApi = {
+  cohorts: CohortApi[];
+  conceptSets: ConceptSet[];
+  valueSets: ValueSetApi[];
 };
 
-type DatasetAccessRequest = {
+export type DatasetAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
+export type DatasetAccessRequestApi = {
+  name: string;
+  researchPurposeStatement: string;
+  datasetRequest: DatasetRequestApi;
+};
+
+export const convertValueSet = (valueSet: ValueSet): ValueSetApi => {
+  return {
+    name: valueSet.domain,
+    values: _.map('name', valueSet.values),
+  };
+};
+
+export const convertCohort = (cohort: Cohort): CohortApi => {
+  return {
+    name: cohort.name,
+    criteriaGroups: _.map(
+      (criteriaGroup) => ({
+        name: criteriaGroup.name,
+        mustMeet: criteriaGroup.mustMeet,
+        meetAll: criteriaGroup.meetAll,
+        count: criteriaGroup.count,
+        criteria: _.map((criteria) => convertCriteria(criteria), criteriaGroup.criteria),
+      }),
+      cohort.criteriaGroups
+    ),
+  };
+};
+
+export const convertCriteria = (criteria: AnyCriteria): AnyCriteriaApi => {
+  const mergeObject = { kind: criteria.kind, name: criteria.name, id: criteria.id };
+  switch (criteria.kind) {
+    case 'range':
+      return <ProgramDataRangeCriteriaApi>_.merge(mergeObject, { low: criteria.low, high: criteria.high });
```

Comment:
I don't think this form `<>` is commonly done in the UI code. I would expect to instead see `as ProgramDataRangeCriteriaApi` here
```suggestion
      return _.merge(mergeObject, { low: criteria.low, high: criteria.high }) as ProgramDataRangeCriteriaApi;
```

assuming that does the same thing